low/low.h

```diff
@@ -106,11 +113,64 @@ void create_kni(uint8_t port, uint8_t core, char *name, struct rte_mempool *mbuf
 	}
 }
 
+int checkRSSPacketCount(struct cPort *port) {
+	return rte_eth_rx_queue_count(port->PortId, port->QueuesNumber-1);
+}
+
+bool changeRSSReta(struct cPort *port, bool increment) {
+	struct rte_eth_dev_info dev_info;
+	memset(&dev_info, 0, sizeof(dev_info));
+	rte_eth_dev_info_get(port->PortId, &dev_info);
+
+	struct rte_eth_rss_reta_entry64 reta_conf[APP_RETA_SIZE_MAX];
+	memset(reta_conf, 0, sizeof(reta_conf));
+
+	// This is required, hanging without this
+	for (int i = 0; i < dev_info.reta_size / RTE_RETA_GROUP_SIZE; i++) {
+		reta_conf[i].mask = UINT64_MAX;
+	}
+
+	if (increment) {
+		if (port->QueuesNumber == dev_info.max_rx_queues) {
+			return false;
+		}
+		port->QueuesNumber++;
+	} else {
+		port->QueuesNumber--;
+	}
+
+        rte_eth_dev_rss_reta_query(port->PortId, reta_conf, dev_info.reta_size);
+
+	// http://dpdk.org/doc/api/examples_2ip_pipeline_2init_8c-example.html#a27
+	for (int i = 0; i < dev_info.reta_size / RTE_RETA_GROUP_SIZE; i++) {
+		for (int j = 0; j < RTE_RETA_GROUP_SIZE; j++) {
+			reta_conf[i].reta[j] = j % port->QueuesNumber;
+		}
+	}
+	rte_eth_dev_rss_reta_update(port->PortId, reta_conf, dev_info.reta_size);
+	return true;
+	// TODO we don't start or stop queues here. Is it right?
```

Comment:
how does it work if you dont know should we resrart queues or not? did u test this function?

Reply:
As it said earlier "This pull request passed teamcity testing. (the same fails like without these changes). So I'm waiting for approves." And also performance was tested, it isn't worse however it is fluky due to above reasons.

Reply:
Got explanations by phone